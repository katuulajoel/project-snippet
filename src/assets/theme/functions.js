export const pxToRem = (pixels) => {
  const value = 1 * (pixels / 16);
  return `${value}rem`;
};

export const functions = {
  pxToRem,
};
