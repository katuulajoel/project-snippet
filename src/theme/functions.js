export const pxToRem = (pixels) => {
  const value = 1 * (pixels / 16);
  return `${value}rem`;
};

const pxToLineHeight = (pixels) => {
  const value = pixels / 16;
  return `${value}rem`;
};

export const functions = {
  pxToRem,
  pxToLineHeight,
};
