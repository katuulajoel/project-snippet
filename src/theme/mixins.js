import { css } from "styled-components";

const clearfix = css`
  &::after {
    display: block;
    clear: both;
    content: "";
  }
`;

const link = (color) => `
    color: ${color};

    &:hover, &:focus, &:active {
        color: ${color};
        text-decoration: none;
    }
`;

const scrollbar = css`
  /* width */
  &::-webkit-scrollbar {
    width: 14px;
    border: 10px solid rgba(0, 0, 0, 0);
    border-top: none;
    border-left: none;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(143, 155, 179, 0.15);
    box-shadow: 0px 4px 8px rgba(62, 72, 87, 0.04);
    background-clip: padding-box;
    border: 10px solid rgba(0, 0, 0, 0);
    border-top: none;
    border-left: none;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    border: 10px solid rgba(0, 0, 0, 0);
    border-top: none;
    border-left: none;
    background-clip: padding-box;
    background-color: #8f9bb3;
    box-shadow: -1px -1px 0px rgba(62, 72, 87, 0.05);
    -webkit-box-shadow: inset -1px -1px 0px rgba(62, 72, 87, 0.05),
      inset 1px 1px 0px rgba(62, 72, 87, 0.05);
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    border: 10px solid rgba(0, 0, 0, 0);
    border-top: none;
    border-left: none;
    background-clip: padding-box;
    background-color: #6b7588;
    cursor: pointer;
  }
`;

const borderBottom = css`
  border-bottom: 1px solid #062e64;
`;

export const mixins = {
  clearfix,
  link,
  borderBottom,
  scrollbar,
};
