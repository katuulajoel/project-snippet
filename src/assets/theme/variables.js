import { pxToRem } from "./functions";

const colors = {
  "dark-blue": "#062E64",
  gray: "#8F9BB3",
  "text-black": "#151A30",
  "text-black-normal": "#3E4857",
  "bg-color": "#f8f8f8",
};

const fontSizes = {
  "font-size-normal": pxToRem(14),
  "font-size-sm": pxToRem(14 * 0.875),
  "font-size-lg": pxToRem(14 * 1.25),
  "font-size-xl": pxToRem(14 * 1.5),
  "font-size-xxl": pxToRem(14 * 1.75),
};

export const variables = {
  colors,
  fontSizes,
};
