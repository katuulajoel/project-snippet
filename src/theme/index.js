import { functions } from "./functions";
import { mixins } from "./mixins";
import { variables } from "./variables";

export default {
  ...variables,
  mixins: mixins,
  functions: functions,
};
