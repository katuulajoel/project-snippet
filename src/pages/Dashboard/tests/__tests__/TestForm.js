import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { ThemeProvider } from 'styled-components';
import TestForm from '../TestForm';
import theme from '../../../../theme';
import {
  // mount,
  shallow,
} from 'enzyme';
// import { render } from '@testing-library/react';

const middlewares = [thunk];

const mockAppState = {
  Invoice: {
    isMakingRequest: {},
    errors: {},
    summary: {},
    list: { data: [], count: 0, next: '', previous: '' },
    invoice: {},
    csv: {},
  },
  TestResults: {
    count: {},
    errors: { fetch: null },
    isFetching: { default: false },
    isSaved: {},
    isSaving: {},
    next: {},
    previous: {},
    results: [],
    selectedFilters: [],
  },
};

export const dummyResult = {
  id: 10,
  comms_check: 'very_good',
  mbti_profile: 'entj',
  iq_test: 100,
  sa_test: 100,
  code_of_conduct: 100,
  coding_tests: [
    {
      skill: 1,
      score: 100,
      id: 12,
      skill_name: 'react',
    },
  ],
  user_obj: {
    id: 16,
    username: 'wd',
    email: 'wd@tunga.io',
    first_name: 'wd',
    last_name: 'dw',
    display_name: 'Wd Dw',
    short_name: 'Wd',
    type: 1,
    image: null,
    is_developer: true,
    is_project_owner: false,
    is_project_manager: false,
    is_staff: false,
    is_admin: false,
    is_pay_admin: false,
    verified: false,
    company: null,
    avatar_url: null,
    can_contribute: false,
    date_joined: '2021-06-22T22:48:41.243936',
    agree_version: 1.2,
    agreed_at: '2021-06-22T22:48:54',
    disagree_version: 0,
    disagreed_at: null,
    payoneer_signup_url: '',
    payoneer_status: 'initial',
    exact_code: '000000000000000016',
    tax_location: 'world',
  },
};

const mockAppStore = (state) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state);
};

describe('Testform tests', () => {
  it('Should match snapshot test', () => {
    const wrapper = shallow(
      <BrowserRouter>
        <Provider store={mockAppStore(mockAppState)}>
          <ThemeProvider theme={theme}>
            <TestForm id="test-form" result={dummyResult} />
          </ThemeProvider>
        </Provider>
      </BrowserRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });

  // it('Should show the label for buttons', () => {
  //   const { container } = render(
  //     <BrowserRouter>
  //       <Provider store={mockAppStore(mockAppState)}>
  //         <ThemeProvider theme={theme}>
  //           <TestForm id="test-form" result={dummyResult} />
  //         </ThemeProvider>
  //       </Provider>
  //     </BrowserRouter>
  //   );
  //   const update = container.querySelector('button.btn-icon');
  //   expect(update.innerHTML).toEqual('');
  //   // expect(update[1].value).toEqual(/Update/i);
  //   // expect(update.values).toMatch(/Update/i);
  // });

  // it('Should trigger dismiss on click', () => {
  //   const onProceedClickMock = jest.fn();
  //   const wrapper = mount(
  //     <BrowserRouter>
  //       <Provider store={mockAppStore(mockAppState)}>
  //         <ThemeProvider theme={theme}>
  //           <TestForm id="test-form" result={dummyResult} proceed={onProceedClickMock} />
  //         </ThemeProvider>
  //       </Provider>
  //     </BrowserRouter>
  //   );
  //   const btn = wrapper.find('button.okay');
  //   expect(btn.innerHTML).toMatch(/Update/i);
  //   btn.simulate('click');
  //   expect(onProceedClickMock).toHaveBeenCalled();
  // });
});
