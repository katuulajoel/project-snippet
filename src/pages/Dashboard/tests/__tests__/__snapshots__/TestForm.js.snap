// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testform tests Should match snapshot test 1`] = `
<DocumentFragment>
  <form
    class="sc-eCApnc gjwEEg"
    id="test-form"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="form-group"
        >
          <label>
            Full name
            <span
              class="sc-ksluID jtelhn"
            >
              <span
                class="label-style"
              >
                *
              </span>
            </span>
          </label>
          <div
            class="sc-pNWdM jOEXEo tag-input"
          >
            <div
              class="form"
            />
            <div
              class="selected"
            >
              <div
                class="item nolabel"
              >
                <div
                  class="sc-fujyAs fhgXi avatar-dash avatar-initials"
                >
                  <div
                    class="avatar avatar-dash avatar-initials"
                  >
                    WD
                  </div>
                </div>
                <span>
                  Wd Dw
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-md-6"
      >
        <div
          class="form-group"
        >
          <label>
            Email address
          </label>
          <div
            class="sc-jSFjdj dqrEkW"
            data-testid="input"
            disabled=""
          >
            <input
              class="form-control"
              data-testid=""
              disabled=""
              id=""
              label="Label"
              placeholder="Email address"
              required=""
              type="text"
              value="wd@tunga.io"
            />
          </div>
        </div>
      </div>
      <div
        class="col-md-12"
      >
        <div
          class="sc-dIsUp dkopAB"
        />
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <label>
          Coding tests
          <span
            class="sc-ksluID jtelhn"
          >
            <span
              class="label-style"
            >
              *
            </span>
          </span>
        </label>
      </div>
      <div
        class="col-md-6"
      >
        <a
          class="sc-crzoAE bOfDmc"
        >
          <i
            class="tg-ic-round-add  "
          />
           Add more
        </a>
      </div>
      <div
        class="col-md-6"
      >
        <div
          class="sc-bqGGPW jvhpmm"
        >
          <div
            class="input-row"
          >
            <div
              class="sc-jrsJWt iekbzz"
            >
              <div
                class="sc-jSFjdj dqrEkW"
                data-testid="input"
                disabled=""
              >
                <input
                  class="form-control"
                  data-testid=""
                  disabled=""
                  id=""
                  label="Label"
                  placeholder="input"
                  type="input"
                  value="react"
                />
              </div>
              <div
                class="sc-jSFjdj dqrEkW"
                data-testid="input"
              >
                <input
                  class="form-control"
                  data-testid=""
                  id=""
                  label="Label"
                  max="100"
                  min="0"
                  placeholder="Score"
                  step="1"
                  type="number"
                  value="100"
                />
              </div>
              <div
                class="item-actions"
              >
                <button
                  class="btn btn-icon sc-iCoGMd fsPrSF  "
                  type="button"
                >
                  <i
                    class="tg-ic-delete-outline tunga-ic-sz-md "
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-md-12"
      >
        <div
          class="sc-dIsUp dkopAB"
        />
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="form-group"
        >
          <label>
            Comms Check
            <span
              class="sc-ksluID jtelhn"
            >
              <span
                class="label-style"
              >
                *
              </span>
            </span>
          </label>
          <div
            class="sc-iqAclL jJanAN"
          >
            <select
              class="form-control  "
            >
              <option
                value=""
              >
                Comms Check
              </option>
              <option
                value="very_good"
              >
                Very good
              </option>
              <option
                value="good"
              >
                Good
              </option>
              <option
                value="pass"
              >
                Pass
              </option>
              <option
                value="poor"
              >
                Bad
              </option>
            </select>
            <i
              class="tg-ic-rounded-keyboard-arrow-down tunga-ic-sz-sm "
            />
          </div>
        </div>
      </div>
      <div
        class="col-md-6"
      >
        <div
          class="form-group"
        >
          <label>
            MBTI Profile
            <span
              class="sc-ksluID jtelhn"
            >
              <span
                class="label-style"
              >
                *
              </span>
            </span>
          </label>
          <div
            class="sc-iqAclL jJanAN"
          >
            <select
              class="form-control  "
            >
              <option
                value=""
              >
                Profiles
              </option>
              <option
                value="estj"
              >
                ESTJ
              </option>
              <option
                value="istj"
              >
                ISTJ
              </option>
              <option
                value="entj"
              >
                ENTJ
              </option>
              <option
                value="intj"
              >
                INTJ
              </option>
              <option
                value="estp"
              >
                ESTP
              </option>
              <option
                value="istp"
              >
                ISTP
              </option>
              <option
                value="entp"
              >
                ENTP
              </option>
              <option
                value="intp"
              >
                INTP
              </option>
              <option
                value="esfj"
              >
                ESFJ
              </option>
              <option
                value="isfj"
              >
                ISFJ
              </option>
              <option
                value="enfj"
              >
                ENFJ
              </option>
              <option
                value="infj"
              >
                INFJ
              </option>
              <option
                value="esfp"
              >
                ESFP
              </option>
              <option
                value="isfp"
              >
                ISFP
              </option>
              <option
                value="enfp"
              >
                ENFP
              </option>
              <option
                value="infp"
              >
                INFP
              </option>
            </select>
            <i
              class="tg-ic-rounded-keyboard-arrow-down tunga-ic-sz-sm "
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="form-group"
        >
          <label>
            IQ Test
            <span
              class="sc-ksluID jtelhn"
            >
              <span
                class="label-style"
              >
                *
              </span>
            </span>
          </label>
          <div
            class="sc-jSFjdj dqrEkW"
            data-testid="input"
          >
            <input
              aria-label="iq-test-input"
              class="form-control"
              data-testid=""
              id=""
              label="Label"
              min="0"
              placeholder="Enter score"
              step="1"
              type="number"
              value="100"
            />
          </div>
        </div>
      </div>
      <div
        class="col-md-6"
      >
        <div
          class="form-group"
        >
          <label>
            SA Test
            <span
              class="sc-ksluID jtelhn"
            >
              <span
                class="label-style"
              >
                *
              </span>
            </span>
          </label>
          <div
            class="sc-jSFjdj dqrEkW"
            data-testid="input"
          >
            <input
              aria-label="sa-test-input"
              class="form-control"
              data-testid=""
              id=""
              label="Label"
              max="100"
              min="0"
              placeholder="SA test"
              step="1"
              type="number"
              value="100"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="form-group"
        >
          <label>
            Code of Conduct
            <span
              class="sc-ksluID jtelhn"
            >
              <span
                class="label-style"
              >
                *
              </span>
            </span>
          </label>
          <div
            class="sc-jSFjdj dqrEkW"
            data-testid="input"
          >
            <input
              aria-label="cc-test-input"
              class="form-control"
              data-testid=""
              id=""
              label="Label"
              max="100"
              min="0"
              placeholder="Enter score"
              step="1"
              type="number"
              value="100"
            />
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
